testing/regress/ecl/embedjs.ecl

```diff
@@ -101,6 +101,32 @@ val[2] = t;
 val;
 ENDEMBED;
 
+r := RECORD
+    UNSIGNED id;
+    STRING name;
+END;
+
+m(unsigned numRows, boolean isLocal = false, unsigned numParallel = 0) := MODULE
+  EXPORT streamed dataset(r) myDataset(unsigned numRows = numRows) := EMBED(javascript : activity, local(isLocal), parallel(numParallel))
+    var numSlaves = __activity__.numSlaves;
+    var numParallel = numSlaves * __activity__.numStrands;
+    var rowsPerPart = (numRows + numParallel - 1) / numParallel;
+    var thisSlave = __activity__.slave;
+    var thisIndex = thisSlave * __activity__.numStrands + __activity__.strand;
+    var first = thisIndex * rowsPerPart;
+    var last = first + rowsPerPart;
+  
+    var names = [ "Gavin", "Richard", "John", "Bart" ];
+    var ds = [ ];
+    while (first < last)
+    {
+        ds.push( { id : first, name: names[first % 4] });
+        first += 1;
+    }
+    ds;
```

Comment:
How does "ds;" magically get returned?

Reply:
I don't remember how the embedded JS works - I just copied the other examples in this file...